ccm_web/server/config.ts

```diff
@@ -0,0 +1,75 @@
+interface ServerConfig {
+  port: number
+}
+
+interface LTIConfig {
+  encryptionKey: string
+  clientID: string
+  platformURL: string
+  authEnding: string
+  tokenEnding: string
+  keysetEnding: string
+}
+
+interface DatabaseConfig {
+  host: string
+  name: string
+  user: string
+  password: string
+}
+
+export interface Config {
+  server: ServerConfig
+  lti: LTIConfig
+  db: DatabaseConfig
+}
+
+const isString = (v: unknown): v is string => typeof v === 'string'
+const isNumber = (v: unknown): v is number => typeof v === 'number'
+
+function validate<T> (
+  key: string,
+  value: unknown,
+  check: (v: unknown) => v is T,
+  fallback?: T
+): T {
+  const errorBase = 'Exception while loading configuration: '
+  if (check(value)) return value
+  if (fallback !== undefined) {
+    console.log(`Value ${String(value)} for ${key} is invalid; using fallback ${String(fallback)}`)
+    return fallback
+  }
+  throw (Error(errorBase + `Value ${String(value)} for ${key} is not valid`))
+}
+
+export function validateConfig (env: Record<string, unknown>): Config | undefined {
+  let server
+  let lti
+  let db
+
+  try {
+    server = {
+      port: validate<number>('PORT', Number(env.PORT), isNumber, 4000)
+    }
+    lti = {
+      encryptionKey: validate<string>('LTI_ENCRYPTION_KEY', env.LTI_ENCRYPTION_KEY, isString, 'LTIKEY'),
```

Comment:
Some how the `LTI_ENCRYPTION_KEY` is commented out in the .env.sample. But regardless if not present looks like you are setting the some string here. I assume that is needed for Prod and non-prod env. Can we generate some random encrypted like we do in Django? so that way we can eliminate messages like below
`Value undefined for LTI_ENCRYPTION_KEY is invalid; using fallback LTIKEY`

Reply:
Yes, we will want to do that (use a better encryption key). I think if you un-comment the line in your `.env` and change it, it should work properly. It gets passed in to the `ltijs` configuration.